auth/index.html

```diff
@@ -1014,88 +1115,37 @@ <h2>Message log</h2>
-          // We do not want to arbitrarily receive messages from all origins.
+          
           window.addEventListener(
             "message",
-            async function (event) {
-              if (
-                event.data &&
-                (event.data["type"] == "INJECT_CREDENTIAL_BUNDLE" ||
-                  event.data["type"] == "INJECT_RECOVERY_BUNDLE")
-              ) {
-                TKHQ.logMessage(
-                  `⬇️ Received message ${event.data["type"]}: ${event.data["value"]}`
-                );
-                try {
-                  await onInjectBundle(event.data["value"]);
-                } catch (e) {
-                  TKHQ.sendMessageUp("ERROR", e.toString());
-                }
-              }
-              if (event.data && event.data["type"] == "STAMP_REQUEST") {
-                TKHQ.logMessage(
-                  `⬇️ Received message ${event.data["type"]}: ${event.data["value"]}`
-                );
-                try {
-                  await onStampRequest(event.data["value"]);
-                } catch (e) {
-                  TKHQ.sendMessageUp("ERROR", e.toString());
-                }
-              }
-              if (event.data && event.data["type"] == "RESET_EMBEDDED_KEY") {
-                TKHQ.logMessage(`⬇️ Received message ${event.data["type"]}`);
-                try {
-                  TKHQ.onResetEmbeddedKey();
-                } catch (e) {
-                  TKHQ.sendMessageUp("ERROR", e.toString());
-                }
-              }
-            },
+            messageEventListener,
             false
           );
 
-          /**
-           * Event handlers to power the recovery and auth flows in standalone mode
-           * Instead of receiving events from the parent page, forms trigger them.
-           * This is useful for debugging as well.
-           */
-          document.getElementById("inject").addEventListener(
-            "click",
-            async function (e) {
-              e.preventDefault();
-              window.postMessage({
-                type: "INJECT_CREDENTIAL_BUNDLE",
-                value: document.getElementById("credential-bundle").value,
-              });
-            },
-            false
-          );
-          document.getElementById("stamp").addEventListener(
-            "click",
-            async function (e) {
-              e.preventDefault();
-              window.postMessage({
-                type: "STAMP_REQUEST",
-                value: document.getElementById("payload").value,
-              });
-            },
-            false
-          );
-          document.getElementById("reset").addEventListener(
-            "click",
-            async function (e) {
-              e.preventDefault();
-              window.postMessage({ type: "RESET_EMBEDDED_KEY" });
-            },
-            false
-          );
+          addDOMEventListeners();
+
+          TKHQ.sendMessageUp("PUBLIC_KEY_READY", targetPubHex);
```

Comment:
I'm worried that if this `sendMessageUp` function is called _before_ `TURNKEY_INIT_MESSAGE_CHANNEL` is received (and thus, the channel set up), then the message will fail to be received on the other end (parent frame) because new SDKs won't listen for raw messages coming from postMessage anymore.

So it seems there's a bit of a race condition, and we should wait to receive "TURNKEY_INIT_MESSAGE_CHANNEL" before broadcasting "PUBLIC_KEY_READY"?

Another way to handle this is to broadcast `PUBLIC_KEY_READY` twice:
* leave this here the way you have it (for old SDKs)
* add a call to `sendMessageUp("PUBLIC_KEY_READY", targetPubHex)` again right after `TKHQ.setParentFrameMessageChannelPort` is called (so that new SDKs receive it as well)

Now that I'm writing this I think that's probably the cleanest option?